tests/Doctrine/Tests/DBAL/DriverManagerTest.php

```diff
@@ -156,30 +157,34 @@ public function testDatabaseUrlPrimaryReplica(): void
         $params = $conn->getParams();
         self::assertInstanceOf(PDOMySQLDriver::class, $conn->getDriver());
 
-        $expected = [
-            'user'     => 'foo',
-            'password' => 'bar',
-            'host'     => 'localhost',
-            'port'     => 11211,
-            'dbname'   => 'baz',
-            'driver'   => 'pdo_mysql',
-            'url'      => 'mysql://foo:bar@localhost:11211/baz',
+        $expectedPrimary = [
+            'user'          => 'foo',
+            'password'      => 'bar',
+            'host'          => 'localhost',
+            'port'          => 11211,
+            'dbname'        => 'baz',
+            'driver'        => 'pdo_mysql',
+            'url'           => 'mysql://foo:bar@localhost:11211/baz?serverVersion=11.2',
+            'serverVersion' => '11.2',
         ];
 
+        $expectedReplica = array_diff_key($expectedPrimary, ['serverVersion' => null]);
```

Comment:
Please [refrain from putting logic in tests](https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html).